docs/development/Offline_mode.md

```diff
@@ -76,3 +76,130 @@ The main problem is deciding what changes should be saved and what changes shoul
 
    
 To accommodate this there might need to add more attributes in the offline and online databases in order to deal with and keep track of the synchronization. Examples could be “last_updated_on”, “created_on”, “deleted_on”, “edited_offline” which are timestamps used to see if data should be synched or not and the “edited_offline” could be a boolean. It is also an option to use UUID with/instead of timestamps to make the synchronization have an unique id. 
+
+
+## Approaches of how to (and not to) implement the offline repository
+
+The [api_client](https://github.com/aau-giraf/api_client) maps json output from the [web-api](https://github.com/aau-giraf/web-api) into models in which the [weekplanner](https://github.com/aau-giraf/weekplanner) uses for displaying data models. Thus, it would be essential to implement the offline repository feature in the api_client. Every model in the api_client implements an abstract class called `Model` which provides a `from_json()` and `to_json()` method for the models to interact with the web-api. 
+
+### NoSQL / Storing plain JSON
+Our first approach was to store the string value of `to_json()` of each model in a local database. With this, you could get all models back from the database with their `from_json()` method. However, this raises a lot of errors when dealing with relational models. Consider this example:
+
+A model `A` has a one-to-many with the model `B`. If an instance of `B`, related to an instance `A`, was to be altered, it would not be altered on `A`.
+
+Thus, storing the plain `to_json()` string values from models is not a valid solution on related models.
+
+
+### In-memory design
+Since mobile devices has become more powerfull and reliable, an in-memory approach is strongly considerable. All models needed for the repository is provided by the api_client making the solution slightly straight forward and performance-efficient (https://en.wikipedia.org/wiki/In-memory_database). 

```

Comment:
```suggestion
Since mobile devices have become more powerful and reliable, an in-memory approach is strongly desirable. All models needed for the repository are provided by the api_client, making the solution slightly straight forward and performance-efficient (https://en.wikipedia.org/wiki/In-memory_database). 

```

Reply:
not sure if you mean "an in-memory approach is strongly **desirable**" or "an in-memory approach **should be considered**"